devlib/trace/perf.py

```diff
@@ -113,29 +143,67 @@ def stop(self):
     # pylint: disable=arguments-differ
     def get_trace(self, outdir):
         for label in self.labels:
-            target_file = self._get_target_outfile(label)
-            host_relpath = os.path.basename(target_file)
-            host_file = _f(os.path.join(outdir, host_relpath))
-            self.target.pull(target_file, host_file)
+            # Sometime perf/simpleperf takes a while to write to .data file after kill signal
+            time.sleep(5)
+            if self.command == 'record':
+                report_command = self._build_perf_report_command(self.report_options, label)
+                self.target.kick_off(report_command, as_root=True)
+                # For larger records, report file takes a while to generate
+                time.sleep(5)
+                self._pull_target_file_to_host(label, 'rpt', outdir)
+            else:
+                self._pull_target_file_to_host(label, 'out', outdir)
 
     def _deploy_perf(self):
         host_executable = os.path.join(PACKAGE_BIN_DIRECTORY,
-                                       self.target.abi, 'perf')
+                                       self.target.abi, self.perf_type)
         return self.target.install(host_executable)
 
+    def _get_target_file(self, label, extension):
+        return self.target.get_workpath('{}.{}'.format(label, extension))
+
     def _build_commands(self):
         commands = []
         for opts, label in zip(self.optionstrings, self.labels):
-            commands.append(self._build_perf_command(opts, self.events, label))
+            if self.command == 'stat':
+                commands.append(self._build_perf_command(opts, self.events, label))
```

Comment:
rename `_build_perf_command` to `_build_perf_stat_command`

Reply:
Fixed